src/Kanvas/Auth/Actions/RegisterUsersAction.php

```diff
@@ -42,35 +43,40 @@ public function execute(): Users
             ['email' => 'required|email|unique:users,email,NULL,id']
         );
 
-        if ($validator->fails()) {
+        // Este es el segundo caso en el que necesitamos conseguir informacion del usuario sin que explote.
+        if ($validator->fails() && UsersAssociatedApps::userOnApp($user = Users::where('email', $this->data->email)->first())) {
```

Comment:
@arfenis why not use https://github.com/bakaphp/kanvas-ecosystem-api/blob/development/src/Kanvas/Users/Models/Users.php#L265 ?